qt/scientific_interfaces/Indirect/Reduction/ISISEnergyTransferModel.cpp

```diff
@@ -216,73 +216,35 @@ std::vector<std::string> IETModel::validatePlotData(IETPlotData const &plotParam
   return errors;
 }
 
-void IETModel::plotRawFile(MantidQt::API::BatchAlgorithmRunner *batchAlgoRunner, InstrumentData const &instData,
-                           IETPlotData const &plotParams) {
-  using Mantid::specnum_t;
-
-  const std::string inputFiles = plotParams.getInputData().getInputFiles();
-
-  int spectraMin = plotParams.getConversionData().getSpectraMin();
-  int spectraMax = plotParams.getConversionData().getSpectraMax();
+std::deque<MantidQt::API::IConfiguredAlgorithm_sptr>
+IETModel::plotRawAlgorithmQueue(InstrumentData const &instData, IETPlotData const &plotParams) const {
+  auto const [rawFile, basename] = parseInputFiles(plotParams.getInputData().getInputFiles());
 
-  std::string rawFile = inputFiles.substr(0, inputFiles.find(',')); // getting the name of the first file
-  std::filesystem::path rawFileInfo(rawFile);
-  std::string name = rawFileInfo.filename().string();
+  auto const data = plotParams.getConversionData();
+  auto const detectorList = createDetectorList(data.getSpectraMin(), data.getSpectraMax());
 
-  auto loadAlg = loadAlgorithm(rawFile, name);
-  if (instData.getInstrument() != "TOSCA") {
-    if (loadAlg->existsProperty("LoadLogFiles")) {
-      loadAlg->setProperty("LoadLogFiles", false);
-    }
-    loadAlg->setPropertyValue("SpectrumMin", std::to_string(spectraMin));
-    loadAlg->setPropertyValue("SpectrumMax", std::to_string(spectraMax));
-  }
-  loadAlg->execute();
-
-  auto inputFromRebin = std::make_unique<Mantid::API::AlgorithmRuntimeProps>();
-  inputFromRebin->setPropertyValue("InputWorkspace", name);
+  return plotRawAlgorithmQueue(rawFile, basename, instData.getInstrument(), detectorList,
+                               plotParams.getBackgroundData());
+}
 
-  std::vector<specnum_t> detectorList;
-  for (specnum_t i = spectraMin; i <= spectraMax; i++)
-    detectorList.emplace_back(i);
+std::deque<MantidQt::API::IConfiguredAlgorithm_sptr>
+IETModel::plotRawAlgorithmQueue(std::string const &rawFile, std::string const &basename,
```

Comment:
When working on the ALFView interface, I came across the same problem - whether it was best to run the BatchAlgorithmRunner from the presenter or from the model. In the end I decided it was better if the presenter owned and called this runner because it is a Qt object, and it felt more natural that the presenter be used as the communication piece between the model and anything else that is Qt related such as the View or BatchAlgorithmRunner. As you say, it also provides a better separation of concerns.

The `ALFInstrumentPresenter` and `ALFAnalysisPresenter` provide an example of something similar to the pattern I was trying to copy a little bit. (The only difference is that they own an `ALFALgorithmManager`, and then this manager owns the BatchAlgorithmRunner)

Reply:
Oh, I've noticed it is a Qt object. Yes, that's make sense